translator/source_code/cotrip_translator.py

```diff
@@ -0,0 +1,421 @@
+from jsonschema import ValidationError
+import logging
+from collections import OrderedDict
+import re
+import dateutil.parser
+
+
+
+
+
+# Translator
+
+
+def wzdx_creator(messages, info=None, unsupported_message_callback=None):
+    if not messages:
+        return None
+   #verify info obj 
+    if not info:
+        info=initialize_info()
+    if not validate_info(info):
+        return None
+    wzd = {}
+    wzd['road_event_feed_info'] = {}
+    # hardcode
+    wzd['road_event_feed_info']['feed_info_id'] = info['feed_info_id']
+    wzd['road_event_feed_info']['update_date'] = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
+    wzd['road_event_feed_info']['publisher'] = 'CDOT'
+    wzd['road_event_feed_info']['contact_name'] = 'Abinash Konersman'
+    wzd['road_event_feed_info']['contact_email'] = 'abinash.konersman@state.co.us'
+    if info['metadata'].get('datafeed_frequency_update', False):
+        wzd['road_event_feed_info']['update_frequency'] = info['metadata'][
+            'datafeed_frequency_update']  # Verify data type
+    wzd['road_event_feed_info']['version'] = '3.0'
+
+    data_source = {}
+    data_source['data_source_id'] = str(uuid.uuid4())
+    data_source['feed_info_id'] = info['feed_info_id']
+    data_source['organization_name'] = info['metadata']['issuing_organization']
+    data_source['contact_name'] = info['metadata']['contact_name']
+    data_source['contact_email'] = info['metadata']['contact_email']
+    if info['metadata'].get('datafeed_frequency_update', False):
+        data_source['update_frequency'] = info['metadata']['datafeed_frequency_update']
+    data_source['update_date'] = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
+    data_source['location_method'] = info['metadata']['wz_location_method']
+    data_source['lrs_type'] = info['metadata']['lrs_type']
+    wzd['road_event_feed_info']['data_sources'] = [data_source]
+
+    wzd['type'] = 'FeatureCollection'
+    sub_identifier = ''.join(random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in
+                             range(6))  # Create random 6 character digit/letter string
+    road_event_id = str(uuid.uuid4())
+    ids = {}
+    ids['sub_identifier'] = sub_identifier
+    ids['road_event_id'] = road_event_id
+
+    wzd['features'] = []
+
+    if not messages.get('alert:Alerts', {}).get('alert:Alert'):
+        return None
+
+    for alert in messages['alert:Alerts']['alert:Alert']:
+        # Parse alert to WZDx Feature    
+        feature = parse_alert(alert, callback_function=unsupported_message_callback)
+        if feature:
+            wzd['features'].append(feature)
+    if not wzd['features']:
+        return None
+    wzd = add_ids(wzd, True)
+    return wzd
+
+
+
+def validate_info(info):
+
+    if ((not info) or (type(info) != dict and type(info) != OrderedDict)):
+        logging.warning('invalid type')
+        return False
+    
+    #### Consider whether this id needs to be hardcoded or generated
```

Comment:
This comment and the one below just seem to be repeats of when we are creating the info object. Let's either update to make sense in this context, or remove.